.gitignore

```diff
@@ -1,42 +1,176 @@
-dist
-*classes
-target/
-build/
-.gradle/
+# Compiled from https://github.com/github/gitignore
+# With Kafka Specific alterations
+
+# Covers Linux OS
+*~
+## KDE directory preferences
+.directory
+## Linux trash folder which might appear on any partition or disk
+.Trash-*
+
+# Covers Windows OS
+## Windows image file caches
+Thumbs.db
+ehthumbs.db
+## Folder config file
+Desktop.ini
+## Recycle Bin used on file shares
+$RECYCLE.BIN/
+## Windows Installer files
+*.cab
+*.msi
+*.msm
+*.msp
+## Windows shortcuts
+*.lnk
+
+# Covers Mac OSX
+.DS_Store
+.AppleDouble
+.LSOverride
+## Thumbnails
+._*
+## Files that might appear in the root of a volume
+.DocumentRevisions-V100
+.fseventsd
+.Spotlight-V100
+.TemporaryItems
+.Trashes
+.VolumeIcon.icns
+## Directories potentially created on remote AFP share
+.AppleDB
+.AppleDesktop
+Network Trash Folder
+Temporary Items
+.apdisk
+
+# Covers Eclipse IDE
+.metadata
+.recommenders
+bin/
+tmp/
+*.tmp
+*.bak
+*.swp
+*~.nib
+local.properties
+.settings/
+.loadpath
+## Eclipse Core
+.project
+## External tool builders
+.externalToolBuilders/
+## Locally stored "Eclipse launch configurations"
+*.launch
+## JDT-specific (Eclipse Java Development Tools)
+.classpath
+## Java annotation processor (APT)
+.factorypath
+
+# Covers IntelliJ IDE
+*.iml
+## Directory-based project format:
+.idea/
+## File-based project format:
+*.ipr
+*.iws
+## Plugin-specific files:
+# IntelliJ
+/out/
+## JIRA plugin
+atlassian-ide-plugin.xml
+
+# Ignore tags created by etags, ctags, gtags (GNU global) and cscope
+TAGS
+.TAGS
+!TAGS/
+tags
+.tags
+!tags/
+gtags.files
+GTAGS
+GRTAGS
+GPATH
+cscope.files
+cscope.out
+cscope.in.out
+cscope.po.out
+
+# Covers SVN
+.svn/
+
+# Covers Java
```

Comment:
See above comment -- these should only show up in directories that are already excluded by other rules.